`make install` fails as the build process by default does not generate docs/netavark-firewalld.7

At commit c4318b525f6f6641c7647e398a53b when I type `make && make install` the latter fails:
```
$ make install
install  -D -m0755 bin/netavark /usr/local/libexec/podman/netavark
make -C docs install
make[1]: Entering directory '/git/containers/netavark/docs'
install -d //usr/local/share/man/man1
install -m 0644 *.1 //usr/local/share/man/man1
install -d //usr/local/share/man/man7
install -m 0644 *.7 //usr/local/share/man/man7
install: cannot stat '*.7': No such file or directory
make[1]: *** [Makefile:24: install] Error 1
make[1]: Leaving directory '/git/containers/netavark/docs'
make: *** [Makefile:99: install] Error 2
```
I have to do `cd docs && make` and afterwards `make install` completes.

* Either `make` should create the documentation under docs/, by calling there `make` (so for instance `make -C docs`); or this should be executed by `make install` before the latter calls `install -m 0644 *.7 //usr/local/share/man/man7`; or in docs/Makefile the `install` goal should depend on the default target⇔detault goal (=goal/target docs).

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -257,10 +257,8 @@ success_task:
     API_URL_BASE: "https://api.cirrus-ci.com/v1/artifact/build/${CIRRUS_BUILD_ID}"
     EXP_BINS: >-
         netavark
-        netavark.debug
         netavark.info
         netavark.aarch64-unknown-linux-gnu
-        netavark.debug.aarch64-unknown-linux-gnu
         netavark.info.aarch64-unknown-linux-gnu
   clone_script: *noop
   bin_cache: *ro_bin_cache

--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,7 @@ else
 endif
 
 .PHONY: all
-all: build
+all: build docs
 
 bin:
 	mkdir -p $@

--- a/Makefile
+++ b/Makefile
@@ -102,6 +102,7 @@ install: $(NV_UNIT_FILES)
 
 .PHONY: uninstall
 uninstall:
+	$(MAKE) -C docs uninstall
 	rm -f $(DESTDIR)$(LIBEXECPODMAN)/netavark
 	rm -f $(PREFIX)/share/man/man1/netavark*.1
 	rm -f ${DESTDIR}${SYSTEMDDIR}/netavark-dhcp-proxy.service

--- a/Makefile
+++ b/Makefile
@@ -107,6 +107,7 @@ uninstall:
 	rm -f $(PREFIX)/share/man/man1/netavark*.1
 	rm -f ${DESTDIR}${SYSTEMDDIR}/netavark-dhcp-proxy.service
 	rm -f ${DESTDIR}${SYSTEMDDIR}/netavark-dhcp-proxy.socket
+	rm -f ${DESTDIR}${SYSTEMDDIR}/netavark-firewalld-reload.service
 
 .PHONY: test
 test: unit integration

--- a/Makefile
+++ b/Makefile
@@ -50,12 +50,10 @@ $(CARGO_TARGET_DIR):
 	mkdir -p $@
 
 .PHONY: build
-build: build_netavark build_proxy_client
-
-.PHONY: build_netavark
-build_netavark: bin $(CARGO_TARGET_DIR)
+build: bin $(CARGO_TARGET_DIR)
 	$(CARGO) build $(release)
 	cp $(CARGO_TARGET_DIR)/$(profile)/netavark bin/netavark$(if $(debug),.debug,)
+	cp $(CARGO_TARGET_DIR)/$(profile)/netavark-dhcp-proxy-client bin/netavark-dhcp-proxy-client$(if $(debug),.debug,)
 
 .PHONY: examples
 examples: bin $(CARGO_TARGET_DIR)
@@ -152,8 +150,3 @@ mock-rpm:
 .PHONY: help
 help:
 	@echo "usage: make $(prog) [debug=1]"
-
-.PHONY: build_proxy_client
-build_proxy_client: bin $(CARGO_TARGET_DIR)
-	$(CARGO) build --bin netavark-dhcp-proxy-client $(release)
-	cp $(CARGO_TARGET_DIR)/$(profile)/netavark-dhcp-proxy-client bin/netavark-dhcp-proxy-client$(if $(debug),.debug,)

--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,8 @@ CARGO ?= cargo
 CARGO_TARGET_DIR ?= targets
 export CARGO_TARGET_DIR  # 'cargo' is sensitive to this env. var. value.
 
+SOURCES = src/** Cargo.toml Cargo.lock
+
 ifdef debug
 $(info debug is $(debug))
   # These affect both $(CARGO_TARGET_DIR) layout and contents
@@ -50,10 +52,13 @@ $(CARGO_TARGET_DIR):
 	mkdir -p $@
 
 .PHONY: build
-build: bin $(CARGO_TARGET_DIR)
+build: bin/netavark
+
+bin/netavark: $(SOURCES) bin $(CARGO_TARGET_DIR)
 	$(CARGO) build $(release)
-	cp $(CARGO_TARGET_DIR)/$(profile)/netavark bin/netavark$(if $(debug),.debug,)
-	cp $(CARGO_TARGET_DIR)/$(profile)/netavark-dhcp-proxy-client bin/netavark-dhcp-proxy-client$(if $(debug),.debug,)
+	cp $(CARGO_TARGET_DIR)/$(profile)/netavark bin/netavark
+	cp $(CARGO_TARGET_DIR)/$(profile)/netavark-dhcp-proxy-client bin/netavark-dhcp-proxy-client
+
 
 .PHONY: examples
 examples: bin $(CARGO_TARGET_DIR)

--- a/contrib/cirrus/runner.sh
+++ b/contrib/cirrus/runner.sh
@@ -19,9 +19,9 @@ _run_noarg() {
 _run_build() {
     # Assume we're on a fast VM, compile everything needed by the
     # rest of CI since subsequent tasks may have limited resources.
-    make all debug=1
     make build_unit  # reuses some debug binaries
     make all  # optimized/non-debug binaries
+    make examples # build the examples binaries needed by the integration tests
     # This will get scooped up and become part of the artifact archive.
     # Identify where the binary came from to benefit downstream consumers.
     cat | tee bin/netavark.info << EOF

--- a/docs/Makefile
+++ b/docs/Makefile
@@ -17,12 +17,17 @@ docs: $(patsubst %.md,%,$(wildcard *.[0-9].md))
 	$(GO) install github.com/cpuguy83/go-md2man/v2@latest
 
 .PHONY: install
-install:
+install: docs
 	install -d ${DESTDIR}/${MANDIR}/man1
 	install -m 0644 *.1 ${DESTDIR}/${MANDIR}/man1
 	install -d ${DESTDIR}/${MANDIR}/man7
 	install -m 0644 *.7 ${DESTDIR}/${MANDIR}/man7
 
+.PHONY: uninstall
+uninstall:
+	rm -f ${DESTDIR}/${MANDIR}/man1/*.1
+	rm -f ${DESTDIR}/${MANDIR}/man7/*.7
+
 .PHONY: clean
 clean:
-	$(RM) *.1
+	$(RM) *.1 *.7

